### DISCUSSION TITLE
aspnet-core-vite/

### DISCUSSION THREAD

**Discussion by giscus** [matteocontrini/blog#5] [Comments]
# aspnet-core-vite/

Vite is a popular build tool for frontends. Here’s how to integrate it in an ASP.NET Core MVC multi-page application.

https://matteosonoio.it/aspnet-core-vite/

**Comment by ameyer117** [comment]
Extremely useful and informative. Thank you!

**Comment by Van-Dame** [comment]
Just a suggestion, you can change `build.outDir` from `'../dist'` to `'../wwwroot/assets'` and keep true to the general ASP.NET project structure.

**Reply by matteocontrini** [reply]
Hey, thanks for the comment. That wasn't possible when I wrote the article, see this issue: https://github.com/Eptagone/Vite.AspNetCore/issues/21

It should be possible now, but I actually don't remember if there were other downsides to that approach. If it works for you, that's a good choice too, yes.

**Comment by ameyer117** [comment]
Just an fyi in case anyone tries this with Vite 5, you'll need to configure `AddViteServices` to look for the manifest.json inside a `.vite` folder now.

**Comment by mehradhn** [comment]
Hi, can u give an example vite mvc with react, please?

**Comment by mrmillross** [comment]
I get `Cannot GET /` when running npm run dev after following this, with .net core 8 and vitejs latest. Everything builds, client and server side. Also if I run the server side from an IDE, it launches a default MVC site with all the styles broken. Is there a 'working' example somewhere on github to see what's missing?

**Reply by matteocontrini** [reply]
You get that error if you try to access the root of the Vite dev server, but that's normal. You should instead run your web application normally and the assets should then be loaded from the dev server. Obviously by default there are no styles so that's not an indication of a problem.

If you think the styles are not being loaded, check the browser console for errors, and the network tab for failed requests.

**Reply by dathoang** [reply]
I get the same error message, and I know that normal because there is no index.html. But main.js and main.css still 404 not found (I've tried open http://localhost:5173/main.js in the browser)
